zip-codes/scripts/search.js

```diff
@@ -11,35 +14,50 @@
           $('#state-select').append('<option>' + a.state + '</option>');
         });
       }
+      initMap(starterMap);
       $('#state-select').on('change', function() {
-        $('#city-select').find('option').remove().end();
-        console.log($(this).val(), 'state value');
+        currState = $(this).val();
+        $('#city-select').empty();
         if ($(this).val()) {
           webDB.execute(
             'SELECT DISTINCT city FROM zips WHERE state="' + $(this).val() + '"' +
             ' ORDER BY city DESC;',
             function(rows){
               if (rows.length){
-                rows.filter(function(dataEntry) {
-                  return $(this).val() === dataEntry.state;
-                })
-                .forEach(function(a){
+                rows.forEach(function(a){
                   $('#city-select').append('<option>' + a.city + '</option>');
+
                 });
               }
             }
           );
         }
       });
+      $('#city-select').on('change', function() {
+        webDB.execute(
+          'SELECT state, latitude, longitude FROM zips WHERE city="' + $(this).val() + '";',
+          function(rows) {
+            if (rows.length){
+              rows.forEach(function(a){
+                if (a.state === currState) {
+                  var newMap = {lat: 0, lng: 0};
+                  newMap = {lat: a.latitude, lng: a.longitude};
```

Comment:
There's no reason to initially set newMap to {lat: 0, lng: 0}. You could simply use line 44 with a var beforehand :)